jobscheduler/scheduler-component-shared/src/java/org/sakaiproject/component/app/scheduler/jobs/LBCSSFolderHideJob.java

```diff
@@ -0,0 +1,84 @@
+package org.sakaiproject.component.app.scheduler.jobs;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.Job;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.sakaiproject.content.api.ContentCollectionEdit;
+import org.sakaiproject.content.api.ContentHostingService;
+import org.sakaiproject.db.api.SqlService;
+import org.sakaiproject.entity.api.ResourceProperties;
+import org.sakaiproject.entity.api.ResourcePropertiesEdit;
+import org.sakaiproject.event.api.UsageSessionService;
+import org.sakaiproject.exception.IdUnusedException;
+import org.sakaiproject.exception.InUseException;
+import org.sakaiproject.exception.PermissionException;
+import org.sakaiproject.exception.TypeException;
+import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.SessionManager;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.util.List;
+
+/**
+ * Job to make LB-CSS folder hidden but contents accessible.
+ * LB-CSS folder should not be visible to access user, this job will hide all existing non-hidden LB-CSS folder.
+ * @author neelam 
+ *
+ */
+@DisallowConcurrentExecution
+public class LBCSSFolderHideJob implements Job {
+
+	private SqlService sqlService;
+	private SessionManager sessionManager;
+
+	private static final Log LOG = LogFactory.getLog(LBCSSFolderHideJob.class);
+	
+	private ContentHostingService contentHostingService;
+
+	public void setSqlService(SqlService sqlService) {
+		this.sqlService = sqlService;
+	}
+
+	public void setSessionManager(SessionManager sessionManager) {
+		this.sessionManager = sessionManager;
+	}
+
+	public void setContentHostingService(ContentHostingService contentHostingService) {
+		this.contentHostingService = contentHostingService;
+	}
+
+	public void execute(JobExecutionContext context) throws JobExecutionException {
+		//set current user in session
+		Session session = sessionManager.getCurrentSession();
```

Comment:
It's good practice to do this in a try/finally block have invalidate the session in the finally. That way you know at the end of the job the session stops.